main.cpp

```diff
@@ -121,357 +163,390 @@ struct CarWash
  4) After you finish defining each type/function, click the [run] button.  Clear up any errors or warnings as best you can.
  */
 
-/*
-Thing 1) Prophet-6
-5 properties:
-    1)  number of oscillators
-    2)  FX on/off
-    3)  filter cutoff amount
-    4)  distortion amount
-    5)  keyboard notes (int)
-3 things it can do:
-    1)  modulate cutoff
-    2)  play sequence
-    3)  turn on FX
-*/
-
 struct Prophet6
 {
-    //1) number of oscillators
     int numOscillators = 12;
-    //2) FX on/off
-    bool fxOnOff = true;
-    //3) filter cutoff amount
+    bool fxOn = true;
     float filterCutoff = 0.5f;
-    //4) distortion amount
     float distortion = 0.01f;
-    //5) keyboard notes (unison)
     int note = 32;
+    int currentNote;
 
     struct Globals
     {
-        //1) transpose value
         int transposeSemiTone = 0;
-        //2) master tune value in cents
         int masterTuneCents = 0; 
-        //3) midi channel
         int midiChannel = 1;
-        //4) midiClock
         int transmitMidiClock = false; 
-        //5) clock port (MIDI or USB)
         bool usingUSBForMidiClock = false; 
 
         void setTransposition(int semitones); 
         void setMasterTune(int cents);
         void setMidiChannel(int channel);
     };
 
-    //1) modulate cutoff
     void modCutoff(float filterCutoff);
-    //2) play sequencer
-    void playSequence(int note);
-    //3) turn on FX
-    bool engageFX(bool fxOnOff);
+    void playSequence();
+    bool engageFX(bool fxOn);
 };
 
-/*
-Thing 2) Tr-808
-5 properties:
-    1)  number of voices
-    2)  step trigger (bool)
-    3)  level amount per voice
-    4)  preset number
-    5)  tempo bpm
-3 things it can do:
-    1)  program sequence
-    2)  adjust tone
-    3)  change the world
- */
+void Prophet6::Globals::setTransposition(int semitones)
+{
+    transposeSemiTone += semitones;
+}
+
+void Prophet6::Globals::setMasterTune(int cents)
+{
+    masterTuneCents += cents;
+}
+
+void Prophet6::Globals::setMidiChannel(int channel)
+{
+    midiChannel = channel;
+}
+
+void Prophet6::modCutoff(float lfo)
+{
+    filterCutoff += lfo;
+}
+void Prophet6::playSequence()
+{}
+
+bool Prophet6::engageFX(bool fxCurrentlyOn)
+{
+    if(fxCurrentlyOn == true)
+    {
+        fxOn = false;
+        return fxOn;
+    }
+    else
+    {
+        fxCurrentlyOn = true;
+        return fxOn;
+    }
```

Comment:
D.R.Y.  
you have `return fxOn;` twice. 
move it outside the if/else
- [ ] revise